TownSuite.ConversionServer/TownSuite.ConversionServer.Utilities.GhostScript/PdfToImageConverter.cs

```diff
@@ -0,0 +1,200 @@
+﻿using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using TownSuite.ConversionServer.Common.Models.Conversions;
+using TownSuite.ConversionServer.Interfaces.Common.Bytes;
+using TownSuite.ConversionServer.Interfaces.Utilities.Converters;
+
+namespace TownSuite.ConversionServer.Utilities.GhostScript
+{
+    public class PdfToImageConverter : IPdfToImageBytesConverter
+    {
+        public const long DEFAULT_MAX_GIGABYTE_SIZE = 10;
+        public const int DEFAULT_MAX_JOB_DURATION_MINUTES = 30;
+        public IByteCount MaxBytes;
+
+        public TimeSpan MaxJobDuration { get; set; }
+
+        public string ExecutableLocation { get; set; }
+
+        public PdfToImageConverter(IOptions<GhostScriptSettings> settings, IByteCountFactory byteCount)
+        {
+            MaxBytes = byteCount.FromGigabytes(settings.Value.MaxGigabytesUpload);
+            MaxJobDuration = TimeSpan.FromMinutes(settings.Value.MaxJobDurationMinutes);
+            ExecutableLocation = settings.Value.ExecutableLocation;
+        }
+
+        public async Task<IEnumerable<byte[]>> Convert(byte[] pdf, CancellationToken cancellationToken = default)
+        {
+            if (pdf.Length <= 0)
+            {
+                return new List<byte[]>();
+            }
+            else if (pdf.Length > MaxBytes.Bytes)
+            {
+                throw new Exception($"PDF SIZE TOO LARGE. Greater than {MaxBytes.Megabytes} megabytes.");
+            }
+
+            var fullCancelToken = mergeTokenWithDefaultDuration(cancellationToken);
+
+            string pdfPath = CreateTempPdfPath();
+
+            await System.IO.File.WriteAllBytesAsync(pdfPath, pdf, fullCancelToken.Token);
+
+            IEnumerable<string> pngPaths = await Convert(pdfPath, fullCancelToken.Token);
+
+            var pngBytes = await GetBytesFromPaths(pngPaths, fullCancelToken.Token);
+
+            CleanFiles(pngPaths, pdfPath);
+
+            return pngBytes;
+        }
+
+        #region "Private Methods"
+        private async Task<IEnumerable<byte[]>> GetBytesFromPaths(IEnumerable<string> filePaths, CancellationToken cancellationToken)
+        {
+            List<byte[]> bytes = new List<byte[]>();
+
+            foreach (string path in filePaths)
+            {
```

Comment:
If it is permitting multi gb files it is better to use a stream instead of byte arrays.